### PR TITLE
Fixes #9657 - merge NICs from compute profile in host create API

### PR DESCRIPTION
**Pull request by tstrachota**
- updated api docs for hosts and interfaces
- host create/update api actions now merge interfaces from compute
  profiles
- NIC type mapping extracted into a separate class
- return full host detail after host update

### CODE REVIEW

**Comment by tstrachota** [comment]
@domcleal could you please tag this one as 1.8 priority?

File: app/controllers/api/v2/hosts_controller.rb [hunk 1]
```diff
@@ -170,6 +182,32 @@ def facts
...
       private
 
+      def merge_interfaces(host)
+        merge = InterfaceMerge.new
+        merge.run(host.interfaces, host.compute_resource.try(:compute_profile_for, host.compute_profile_id))
+      end
+
+      def host_attributes(params)
+        return {} if params.nil?
+
```
Comment (ares) [inline app/controllers/api/v2/hosts_controller.rb:191]: Could we deep_clone params here? We shouldn't change params values in this method, we should just return new hash with correctly nested values I think. The reason is that something else might expect params in their original form...

File: test/unit/interface_type_mapper_test.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+
+class InterfaceTypeMapperTest < ActiveSupport::TestCase
+  def setup
+    @mapper = InterfaceTypeMapper
+  end
+
```
Comment (ares) [inline test/unit/interface_type_mapper_test.rb:7]: maybe one more test for DEFAULT_TYPE for nil?

**Comment by tstrachota** [comment]
@ares rebased and updated according to your comments

**Comment by ares** [comment]
@tstrachota tests are [failing](http://ci.theforeman.org/job/test_develop_pr_core/4327/database=postgresql,ruby=1.9.3/#showFailuresLink)

**Comment by tstrachota** [comment]
@ares per the irc discussion I removed the `_destroy` check. I also added extended interfaces' rabl with `:primary, :provision` fields as they were missing.

**Comment by ares** [comment]
ready to merge pending Mr Jenkins approval

**Comment by ares** [comment]
@tstrachota oops, Mr. Jenkins is not happy

**Comment by tstrachota** [comment]
@ares yes, I forgot `_delete: 1` in the tests which fails because the field name is actually `_destroy` as you've mentioned earlier today. Removed now, sorry.

**Comment by ares** [comment]
Merged as cfd1c413362813f24bffd62c633bf6473e051740, thanks @tstrachota!
